doc/manual/syntax.md

```diff
@@ -0,0 +1,221 @@
+
+Examples:
+```
+> true && false
+false
+
+> false || true
+true
+
+> ! true
+false
+```
+
+### Strings
+
+Nickel can work with sequences of characters, or strings. 
+Strings are enclosed by `" ... "` for a single line string or by `m#" ... "#m` for a multiline string.
+They can be concatenated with the operator `++`.
+Strings must be UTF-8 valid.
+
+Examples:
+```
+> "Hello, World!"
+"Hello, World!"
+
+> m#"Well, if this isn't a multiline string?
+Yes it is, indeed it is"#m
+"Well, if this isn't a string?
+Yes it is, indeed it is"
+
+> "Hello" ++ "World"
+"HelloWorld"
+```
+
+## Equality Operators
+
+We have seen that the operators `==` and `!=` works for comparing numbers, but they can actually compare any values together. Two values are equal if they have the same _type_ and the same _value_.
+
+Examples:
+```
+> 1 == 1
+true
+
+> 5 == 5.0
+true
+
+> "Hello" == "Hello"
+true
+
+> "Hello" != "World"
+true
+
+> 5 == "Hello"
+false
+
+> true == "true"
+false
+```
+
+
+## Composite types
+
+### List
+Lists allows grouping of values into sequences. They are delimited by `[` and `]`, and elements are separated with `,`.
+
+Examples:
+```
+[1, 2, 3]
+["Hello", "World"]
+[1, true, "true"]
+[]
+```
+
+Lists can be concatenated with the operator `@`:
+```
+> [1] @ [2, 3]
+[ 1, 2, 3 ]
+```
+
+### Record
+Records are key-value storage. They are delimited with by `{` and `}`, elements are separated by `,`.
+key-value elements are noted as `key = value`.
+The keys can be either identifiers, or strings. Values can be of any type.
+Elements inside a record are unordered.
+
+Examples:
+```
+{}
+{a = 3}
+{my_id_n5 = "my id number 5", my_id_n4 = }
+{"5" = 5, six = 6}
+```
+
+A record cannot contain the same key twice, or the values must be identical (see section about _merge_).
+```
+> {a = 3, a = 3}
+{ a = 3 }
+
+> {a = 3, "a" = 3}
+{ a = 3 }
+```
+
+Accessing a record field can be done using the `.` operator :
+```
+> { a = 1, b = 5 }.a
+1
+
+> { a = 1 }.b
+error: Missing field
+
+> { "1" = "one" }."1"
+"one"
+```
+
+## Statements
+
+### If-Then-Else
+This statement allows conditional branching in your code. You can use it as `if <bool expr> then <expr> else <expr>`. Both expressions in the `then` and `else` branches are lazily evaluated, and must have the same type.
+
+Examples:
+```
+> if true then "TRUE :)" else "false :("
+"TRUE :)"
+
+> if false then "Not this one" else "This one"
+"This one"
+
+> if "forty-two" == 42 then "equal?" else "unequal"
+"unequal"
+```
+
+### Let-In
+Let-in allows the binding of an expression. It is used as `let <ident> = <expr> in <expr>`.
+
+Examples:
+```
+> let r = { a = "a", b = "b" } in r.a
+"a"
+
+> let inner = { inside = true } in let outer = { outside = inner.inside } in outer.outside
+true
+
+> let a = 1 in let b = 2 in a + b
+3
+```
+
+The identifier in the `let` part can be typed (more about typing in the dedicated document):
+```
+> let a : Num = 6 in 6 * 7
+42
+```
```

Comment:
I'm not sure this should be in the let section. The thing is, any value can have a type annotation `5 : Num`. Then, any *binding* (record fields and let) have this syntactic sugar `let foo : Type = ...` or `{foo : Type = ...}` as syntactic sugar for respectively `let foo = (... : Type)` and `{foo = (... : Type)}`. It could be in a separate section "Annotations" or something, together with metavalues.

Reply:
Moved typing to its own section